### PR TITLE
Update to newer version of JAX

### LINKED ISSUE #146: binomial/multinomial tests fail with the new version of jax
Start with jax version 0.1.28, as @neerajprad observed, tests for Binomial/Multinomial failed. I am not sure what is the reason for it because except for the tests, these distributions work well in notebooks.

### PR DESCRIPTION
**Pull request by fehiepsi**
Fixes #146

Originally, this PR is to implement TruncatedCauchy (to port local global trend to new distribution API) but it seems that we can fix #146 by casting device array to onp.array. Given that it is more urgent (to make our tests faster), I push the fix instead.

In addition, I also add default args to our distribution interface.

### CODE REVIEW

File: test/test_distributions.py [hunk 1]
```diff
@@ -216,27 +216,30 @@ def test_sample_gradient(jax_dist, sp_dist, params):
         pytest.skip('{} not reparametrized.'.format(jax_dist.__name__))
 
     dist_args = [p.name for p in inspect.signature(jax_dist).parameters.values()]
+    params_dict = dict(zip(dist_args[:len(params)], params))
+    nonrepara_params_dict = {k: v for k, v in params_dict.items()
+                             if k not in jax_dist.reparametrized_params}
+    repara_params = tuple(v for k, v in params_dict.items()
+                          if k in jax_dist.reparametrized_params)
```
Comment (fehiepsi) [inline test/test_distributions.py]: This is to only take `grad` w.r.t. reparameterized params.

File: numpyro/distributions/continuous.py [hunk 2]
```diff
@@ -468,6 +503,8 @@ class Pareto(TransformedDistribution):
     arg_constraints = {'alpha': constraints.positive, 'scale': constraints.positive}
     support = constraints.real
 
+    # FIXME: should we use `concentration` and `scale` for consistence with other distributions
+    # tensorflow uses `concentration` and `scale` though
```
Comment (fehiepsi) [inline numpyro/distributions/continuous.py]: I think it is better if we can swap `alpha` and `scale` (which makes Pareto closer to scipy/tensorflow than pytorch). There we can also set default parameter `scale=1`.

File: numpyro/distributions/util.py [hunk 3]
```diff
@@ -206,6 +206,7 @@ def _binomial(key, p, n=1, shape=()):
 
 
 def binomial(key, p, n=1, shape=()):
+    n = device_get(n)
```
Comment (fehiepsi) [inline numpyro/distributions/util.py]: I think that `static_args` does not like DeviceArray anymore. Anyway this is no-op if n is numpy array so I guess things should be fine.

File: numpyro/distributions/continuous.py (hunk 2 above)
Comment (neerajprad) [reply numpyro/distributions/continuous.py]: Seems reasonable to do.

File: numpyro/distributions/util.py (hunk 3 above)
Comment (neerajprad) [reply numpyro/distributions/util.py]: Hmm..that's quite strange. Is this intended?

File: numpyro/distributions/continuous.py [hunk 4]
```diff
@@ -200,13 +200,48 @@ def __init__(self, df, validate_args=None):
         super(Chi2, self).__init__(0.5 * df, 0.5, validate_args=validate_args)
 
 
+class GaussianRandomWalk(Distribution):
+    arg_constraints = {'num_steps': constraints.positive_integer, 'scale': constraints.positive}
+    support = constraints.real
+    reparametrized_params = ['scale']
```
Comment (neerajprad) [inline numpyro/distributions/continuous.py]: Nice to see that this is working!

File: test/test_distributions.py [hunk 5]
```diff
@@ -216,27 +216,30 @@ def test_sample_gradient(jax_dist, sp_dist, params):
...
     rng = random.PRNGKey(0)
 
     def fn(args):
-        return np.sum(jax_dist(*args).sample(key=rng))
+        args_dict = dict(zip(jax_dist.reparametrized_params, args))
+        return np.sum(jax_dist(**args_dict, **nonrepara_params_dict).sample(key=rng))
 
-    actual_grad = jax.grad(fn)(params)
-    assert len(actual_grad) == len(params)
+    actual_grad = jax.grad(fn)(repara_params)
```
Comment (neerajprad) [inline test/test_distributions.py]: +1.

File: numpyro/distributions/util.py (hunk 3 above)
Comment (fehiepsi) [reply numpyro/distributions/util.py]: I am not sure about that. We can file an issue to JAX devs if this makes regression in performance.

**Pull request merged**